src/native/mac/qgsmacnative.mm

```diff
@@ -0,0 +1,36 @@
+/***************************************************************************
+    qgsmacnative.cpp - abstracted interface to native Mac objective-c
+                             -------------------
+    begin                : January 2014
+    copyright            : (C) 2014 by Larry Shaffer
+    email                : larrys at dakotacarto dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmacnative.h"
+
+#include <Cocoa/Cocoa.h>
+
+QgsMacNative::~QgsMacNative()
+{
+}
+
+const char* QgsMacNative::currentAppLocalizedName()
+{
+  return [[[NSRunningApplication currentApplication] localizedName] UTF8String];
+}
+
+void QgsMacNative::currentAppActivateIgnoringOtherApps()
+{
+  // valid for Mac OS X >= 10.6
+  [[NSRunningApplication currentApplication] activateWithOptions:
+    (NSApplicationActivateAllWindows | NSApplicationActivateIgnoringOtherApps)];
```

Comment:
I guess it depends upon the use of the abstracted call. If it is run from _within_ QGIS, e.g. `Window -> Bring all to Front` menu action, or in drag/drop operations, then this is expected behavior. If it is called by a background process in QGIS, while the user is in another application space, then, yes, that's bad form.

Currently, this is only used to bring the app windows to front from within the app, for which I have not checked yet to see if Qt5 offers a solution over Qt4.